src/DataFrame/Operations/Merge.hs

```diff
@@ -27,9 +25,9 @@ instance Semigroup D.DataFrame where
                     pure $ D.insertColumn name col df
                 | otherwise =
                     let
-                        numColumnsA = (fst $ D.dimensions a')
-                        numColumnsB = (fst $ D.dimensions b')
-                        numColumns = max numColumnsA numColumnsB
+                        numColumnsA = snd $ D.dimensions a'
```

Comment:
The only change that's something else than "removing unused bindings".
The name of this binding suggests you care about column count (which is the seconds component of the dimetnions tuple (rows, columns). If that's the case, `snd` should be used here instead of `fst`.